tests/e2e/volumecreate_shared_test.go

```diff
@@ -0,0 +1,226 @@
+	}
+}
+
+func (s *VolumeCreateSharedTestSuite) TearDownTest(c *C) {
+	volList := strings.Join(s.volumeList, " ")
+
+	if volList != "" {
+		out, err := dockercli.DeleteVolume(s.config.DockerHosts[0], volList)
+		c.Assert(err, IsNil, Commentf(out))
+	}
+
+	// clean the list of volumes created
+	s.volumeList = s.volumeList[:0]
+}
+
+var _ = Suite(&VolumeCreateSharedTestSuite{})
+
+// create volume and do valid/invalid assertion
+func (s *VolumeCreateSharedTestSuite) createVolCheck(name, option string, valid bool, c *C) {
+	var out string
+	var err error
+
+	if option == "" {
+		out, err = dockercli.CreateSharedVolume(s.config.DockerHosts[0], name)
+	} else {
+		out, err = dockercli.CreateSharedVolumeWithOptions(s.config.DockerHosts[0], name, option)
+	}
+
+	// if creation is successful, add it to the list so that it gets cleaned later
+	if err == nil {
+		s.volumeList = append(s.volumeList, name)
+	}
+
+	if valid {
+		// positive test case
+		c.Assert(err, IsNil, Commentf(out))
+	} else {
+		// negative test case
+		c.Assert(err, Not(IsNil), Commentf(out))
+		c.Assert(strings.HasPrefix(out, dockerclicon.ErrorVolumeCreate), Equals, true, Commentf(out))
+	}
+}
+
+// loop over volume names to parallely create volumes
+func (s *VolumeCreateSharedTestSuite) parallelCreateByName(names []string, valid bool, c *C) {
+	var wg sync.WaitGroup
+	for _, volName := range names {
+		wg.Add(1)
+		go func(name string) {
+			defer wg.Done()
+			s.createVolCheck(name, "", valid, c)
+		}(volName)
+	}
+	wg.Wait()
+}
+
+// loop over volume options to parallely create volumes
+func (s *VolumeCreateSharedTestSuite) parallelCreateByOption(options []string, valid bool, c *C) {
+	var wg sync.WaitGroup
+	for _, volOption := range options {
+		wg.Add(1)
+		go func(option string) {
+			defer wg.Done()
+			volName := inputparams.GetUniqueVolumeName("option")
+			s.createVolCheck(volName, option, valid, c)
+		}(volOption)
+	}
+	wg.Wait()
+}
+
+func (s *VolumeCreateSharedTestSuite) accessCheck(hostIP string, volList []string, c *C) {
+	isAvailable := verification.CheckVolumeListAvailability(hostIP, volList)
+	c.Assert(isAvailable, Equals, true, Commentf("Volume %s is not available after creation", volList))
+}
+
+// Valid volume names test
+func (s *VolumeCreateSharedTestSuite) TestValidName(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	s.parallelCreateByName(s.validVolNames(), true, c)
+	s.accessCheck(s.config.DockerHosts[0], s.volumeList, c)
+
+	misc.LogTestEnd(c.TestName())
+}
+
+// Invalid volume names test
+func (s *VolumeCreateSharedTestSuite) TestInvalidName(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	s.parallelCreateByName(invalidVolNameList, false, c)
+
+	misc.LogTestEnd(c.TestName())
+}
+
+// Valid volume creation options
+// 1. size 10gb
+// 2. disk format (thin, zeroedthick, eagerzeroedthick)
+// 3. attach-as (persistent, independent_persistent)
+// 4. fstype ext4 for linux
+// 5. access (read-write, read-only)
```

Comment:
We should only support size for now.
The other options are not valid for shared plugin.